GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.UploadObject.cs

```diff
@@ -0,0 +1,197 @@
+                source, options, progress);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage asynchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public Task<Object> UploadObjectAsync(string bucket, string objectName, string contentType, Stream source)
+        {
+            return UploadObjectAsync(bucket, objectName, contentType, source,
+                options: null, cancellationToken: CancellationToken.None, progress: null);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage asynchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <param name="options">Additional options for the upload. May be null, in which case appropriate
+        /// defaults will be used.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <param name="progress">Progress reporter for the upload. May be null.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public Task<Object> UploadObjectAsync(
+            string bucket,
+            string objectName,
+            string contentType,
+            Stream source,
+            UploadObjectOptions options,
+            CancellationToken cancellationToken,
+            IProgress<IUploadProgress> progress)
+        {
+            ValidateBucket(bucket);
+            objectName.CheckNotNull(nameof(objectName));
+            contentType.CheckNotNull(nameof(contentType));
+            return UploadObjectAsync(new Object { Bucket = bucket, Name = objectName, ContentType = contentType },
+                source, options, cancellationToken, progress);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage asynchronously, from a specified stream.
+        /// </summary>
+        /// <param name="destination">Object to create or update. Must not be null, and must have the name and bucket populated.</param>
```

Comment:
We're also using the `ContentType` from the `Object` therefore we should also say that the `ContentType` must not be `null`.